include/unifex/async_scope.hpp

```diff
@@ -170,6 +170,14 @@ struct async_scope {
     }
   }
 
+  template (typename Fun, typename Scheduler)
+    (requires callable<Fun> && scheduler<Scheduler>)
+  void spawn_with(Fun&& fun, Scheduler&& scheduler) {
```

Comment:
What do you think of taking the callable second?  Code that looks like this reads better, I think:
```c++
scope.spawn_with(scheduler, []() noexcept {
  // something very long that makes the
  // lambda body break lines
});
```
while this is harder to read:
```c++
scope.spawn_with([]() noexcept {
  // something very long that makes the
  // lambda body break lines
}, scheduler);
```

Reply:
My beef with this suggestion is that `on` takes the scheduler second, and `spawn_on` also takes the scheduler second. I would dislike the inconsistency more that I dislike putting the scheduler after the lambda.

Reply:
Maybe on should change, though. The only good reason to have the Sender first is for piping, but on(sched, blah | blah | blah) might actually be easier to read, knowing where something is going before what it is you are sending there.

Reply:
I agree that we shouldn't support pipe syntax with on. It's seriously confusing.

I'd like to hear from @lewissbaker and @kirkshoop before I make that change, though.

Reply:
For another data point, `via` takes the scheduler first. At the very least, we're being inconsistent here.

Reply:
via() should really be taking a scheduler rather than the schedule-sender.
We have finally() which does the equivalent of a via() but takes a sender instead of a scheduler.